card/mod/03_machines/lib/card/machine.rb

```diff
@@ -178,7 +178,7 @@ def machine_output_path
     end
 
     def ensure_machine_output
-      if !output = fetch(:trait => :machine_output) or !output.selected_content_action_id
+      if !output = fetch(:trait => :machine_output) #or !output.selected_content_action_id
```

Comment:
I suspect the above is actually needed.  But I think we can do this more efficiently (see comment re selected actions below)